### ISSUE TITLE
[Feature Request] Per field type apperance in the revisions history view.

### ISSUE THREAD

**Issue by mamarmite** [Laravel-Backpack/revise-operation#12]
# Feature Request (more of a concept presentation before beginning implementing it)

### What's the feature you think this package should have?
Have per field type view as we show revisions. Like a view for text, longtext, dates, dates range, images, etc.

### Have you already implemented a prototype solution, for your own project?
Nope, I'm in the process of designing the logic, the algorithm.

Pretty like the fields algorithm of the CRUD, it would print changes with the field appearance.
Like with a fields folder with field blade's views.

![Exemple d'une rangé simple](https://user-images.githubusercontent.com/820776/107526019-6983cd80-6b85-11eb-9dcd-2a46a50a2a85.png)

For now in my current project, I only check the length of froms and tos, and show the big space (by default with the package) and that view if it's a small changes, like a date range, a varchar, etc.

### Do you see this as a core feature or an add-on?

I would see it as a core feature, but I'm in no place to say that. I think it should follow the same design as the crud's field. Maybe overkill ?

**Comment by promatik** [comment]
Hi @mamarmite this looks like a great improvement for `revise-operation`.

Sure it would be better to have a visual representation of a checkbox, instead of;
![image](https://user-images.githubusercontent.com/1838187/107891836-05f8f780-6f19-11eb-8123-9a93b3b2945d.png)

In some cases it may not make sense to have a visual representation of the field, but in many it does, and yes, there could be a system where we check if there is a blade to represent that change, if it doesn't, we print the raw text.

I like the idea 👌
If you have the time, PR's are welcome 🙌
I'll move this to v4.2 since we are now in feature freeze mode, and I'll let @tabacitu and @pxpm give their opinions on this.

**Comment by mamarmite** [comment]
I've accomplish some advancement in that feature, but still in project context. I need to test it in 4.2 and implement it as in the package scope.
I don't know yet, when I'll be able to finished this. This spring I hope. 

If someone have time, I can push all the things done at this point, to start from or to refactor from.

**Comment by tabacitu** [comment]
I love the mockup/screenshot @mamarmite . Totally agree that would be better than what we have now, in most cases I think the value is very short, and it doesn't make sense to have so much visual clutter for it. Should test with big values too (textarea etc)

I think you meant `columns`, though, right? That the before/after value is shown in a different way depending on the `column` type (not field type). I'm asking that because it makes more sense to me to use the `columns`:
- they're not editable, just for show
- they can be inferred pretty easily with `CRUD::setFromDb()` (though it's not recommended)
- most likely they're already defined in a different operation (like List of Show) so Revise can just take the configured column types from there

**Comment by mamarmite** [comment]
Yes, I meant column, I used the field because in my mind they are the only modifiable elements, which can have revision to it.
Just to push that in this issue, I implemented (only my project scope for now) to the style as shown in #6 
With bootstrap's classes to mimic the milestone style without having to load another structure to the css.

Good idea to use the configured column there.
